Add Java-Client-API to Katacoda Scenario (#5)
cyberark/katacoda-scenarios: java-client-api-tutorial/step1.md

> # Welcome to: Using the Java client library to access the Conjur API!

Reviewer: ```suggestion
# Welcome to: Using Java Applications with Conjur in Kubernetes!
```

> Welcome to Conjur's tutorial of how to use our Java client library to access our Conjur API.

Reviewer: ```suggestion
Welcome to Conjur's tutorial of how to deploy a Java application to Kubernetes and configure it to communicate with Conjur.
```

> In this tutorial, you will learn how to run a local Java application that connects to the Conjur API given an access token provided by a Kubernetes authenticator sidecar on a local Kubernetes cluster using Minikube, login to the Conjur CLI, authenticate and set policies, and fetch a secret.

Reviewer: ```suggestion
In this tutorial, you will learn how to deploy a Java application in Kubernetes that is configured to communicate with a Conjur server. The application will be deployed with the Conjur Kubernetes authenticator client sidecar, which manages the Conjur authentication process and ensures the app always has access to a valid Conjur access token. The application uses the Conjur Java client library to fetch a secret.
```

## Overview

> Applications and application developers should be incapable of leaking secrets.
> 

Reviewer: ```suggestion
```
I don't think we're achieving this goal with this tutorial, fwiw

> In order to achieve this goal, you will play three roles in this tutorial:

Reviewer: ```suggestion
In this tutorial, you will play three roles:
```

1. An **Infrastructure Engineer**, who will deploy an instance of Conjur

> 2. A **Security Admin** who handles secrets, and has sole access to their secrets

Reviewer: ```suggestion
2. A **Security Admin** who manages application access control, to ensure apps can only access the resources they're supposed to
```

> 3. An **Application Developer** who wants to connect to Conjur to access secrets to be used in their application.

Reviewer: ```suggestion
3. An **Application Developer** who wants to deploy their Java application to Kubernetes, and make sure it has the credentials to connect to the resources it needs
```
you've been an app developer. when you're building your app, is your primary goal ever "connecting to a secrets service"?

Specifically, we will:

**As the Infrastructure Engineer:**

> 1. Create an instance of Conjur OSS and a PostgreSQL database

Reviewer: ```suggestion
1. Deploy Conjur OSS, its PostgreSQL database, and Nginx for TLS termination in a pod in its own namespace
```

**As the Security Admin**

> 1. Creates the application identity

Reviewer: ```suggestion
1. Create a new application identity in Conjur
```

> 2. Provides the developer with instructions on how to authenticate
> 3. Stores the secrets

Reviewer: ```suggestion
2. Stores the secrets the application will need, and ensure the app will be able to access them
3. Provides the developer with Conjur configuration instructions for their application
```

**As the Application Developer:**

> 1. Create an application to use the token and Java Client to retrieve the given secrets
> 2. Update their application manifest

Reviewer: ```suggestion
1. Update the application manifest to include the Conjur authenticator client sidecar and a volume mount for the Conjur access token
2. Update the application to use the Java client library and the sidecar-supplied Conjur access token to retrieve the credentials the app needs from Conjur
3. Deploy the application, and verify it's up and running!
```
